envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactory.kt

```diff
@@ -87,8 +38,63 @@ class LuaFilterFactory(incomingPermissionsProperties: IncomingPermissionsPropert
             .setTypedConfig(Any.pack(Lua.newBuilder().setInlineCode(ingressClientNameHeaderScript).build()))
             .build()
 
+    fun ingressScriptsMetadata(group: Group): Metadata {
+        return Metadata.newBuilder()
+            .putFilterMetadata("envoy.filters.http.lua",
+                Struct.newBuilder()
+                    .putFields("client_identity_headers",
+                        Value.newBuilder()
+                            .setListValue(ListValue.newBuilder()
+                                .addAllValues(
+                                    incomingPermissionsProperties.clientIdentityHeaders
+                                        .map { Value.newBuilder().setStringValue(it).build() }
+                                )
+                                .build()
+                            ).build()
+                    )
+                    .putFields("request_id_headers",
+                        Value.newBuilder()
+                            .setListValue(ListValue.newBuilder()
+                                .addAllValues(
+                                    incomingPermissionsProperties.requestIdentificationHeaders
+                                        .map { Value.newBuilder().setStringValue(it).build() }
+                                )
+                                .build()
+                            ).build()
+                    )
+                    .putFields(
+                        "trusted_client_identity_header",
+                        Value.newBuilder()
+                            .setStringValue(trustedClientIdentityHeader)
+                            .build()
+                    )
+                    .putFields(
+                        "san_uri_lua_pattern",
+                        Value.newBuilder()
+                            .setStringValue(
+                                SanUriMatcherFactory(incomingPermissionsProperties.tlsAuthentication)
```

Comment:
Ta zmiana powoduje, że `ingressScriptsMetadata` będzie teraz generowane dla dla każdego envoya oddzielnie. Do tej pory było generowane raz dla wszystkich envoyów. 

Może warto więc przyjrzeć się, czy nie dzieje się tu coś kosztownego, co nie jest zmienne per envoy, a więc można by to tworzyć tylko raz.

`SanUriMatcherFactory` wydaje się kandydatem - nie jest zależny od grupy, a w środku są tworzone np. Regexy.

Reply:
W zasadzie to nie `SanUriMatcherFactory()`, ale nawet `sanUriWildcardRegexForLua` można liczyć tylko raz

Reply:
tak, tylko za bardzo nie widze innej możliwości by w filtrach mieć dostęp do serviceName instancji do której przyszedł request, będzeimy sprawdzać, jeśli wydajnościowo będą problemy, to wrócimy do stałego obiektu dla metadanaych filtrów

Reply:
Ale chodzi o wyciągnięcie do stałej tylko `sanUriWildcardRegexForLua`, czemu tego nie można zrobić?

Reply:
aa, ok, juz zmieniam